processing/src/main/java/io/druid/query/aggregation/BufferAggregator.java

```diff
@@ -110,4 +110,16 @@
    * Release any resources used by the aggregator
    */
   void close();
+
+  /**
+   * Relocates any cached objects.
+   * <b>Implementations must not change the position, limit or mark of the given buffer</b>
+   *
+   * @param oldPostition old position of an item in old ByteBuffer.
+   * @param newPosition  new position of an item in new ByteBuffer.
+   * @param newBuffer    ByteBuffer to be used.
+   */
+  default void relocate(int oldPostition, int newPosition, ByteBuffer newBuffer)
```

Comment:
Might be more efficient to pass to this method `newBuffer` and two parallel sequences (lists, iterators, suppliers, whatever) of oldPositions and newPositions. This would allow to optimize `relocate()` implementation in `SketchBufferAggregator` specifically, because it won't access `nmCache` on each update.